src/Qoollo.PerformanceCounters/WinCounters/Counters/WinDeltaCounter.cs

```diff
@@ -0,0 +1,232 @@
+        /// Конструктор WinNumberOfItemsCounter
+        /// </summary>
+        /// <param name="name">Имя счётчика</param>
+        /// <param name="description">Описание счётчика</param>
+        /// <param name="info">Информация о функционировании</param>
+        internal WinDeltaCounter(string name, string description, WinCountersWorkingInfo info)
+            : base(name, description)
+        {
+            _state = WinCounterState.Created;
+            _info = info;
+        }
+
+        /// <summary>
+        /// Уменьшить значение на 1
+        /// </summary>
+        /// <returns>Текущее значение или Counter.FailureNum</returns>
+        public override long Decrement()
+        {
+            var counterCpy = _winCounter;
+            if (counterCpy == null)
+                return Counter.FailureNum;
+
+            return counterCpy.Decrement();
+        }
+
+        /// <summary>
+        /// Уменьшить значение на "value"
+        /// </summary>
+        /// <param name="value">Значение, на которое уменьшаем</param>
+        /// <returns>Текущее значение или Counter.FailureNum</returns>
+        public override long DecrementBy(long value)
+        {
+            var counterCpy = _winCounter;
+            if (counterCpy == null)
+                return Counter.FailureNum;
+
+            return counterCpy.IncrementBy(-value);
+        }
+
+        /// <summary>
+        /// Увеличить значение на 1
+        /// </summary>
+        /// <returns>Текущее значение или Counter.FailureNum</returns>
+        public override long Increment()
+        {
+            var counterCpy = _winCounter;
+            if (counterCpy == null)
+                return Counter.FailureNum;
+
+            return counterCpy.Increment();
+        }
+
+        /// <summary>
+        /// Увеличить значение на "value"
+        /// </summary>
+        /// <param name="value">Значение, на которое увеличиваем</param>
+        /// <returns>Текущее значение или Counter.FailureNum</returns>
+        public override long IncrementBy(long value)
+        {
+            var counterCpy = _winCounter;
+            if (counterCpy == null)
+                return Counter.FailureNum;
+
+            return counterCpy.IncrementBy(value);
+        }
+
+        /// <summary>
+        /// Задать значение
+        /// </summary>
+        /// <param name="value">Новое значение</param>
+        public override void SetValue(long value)
+        {
+            var counterCpy = _winCounter;
+            if (counterCpy == null)
+                return;
+
+            counterCpy.RawValue = value;
+        }
+
+        /// <summary>
+        /// Получает текущее значение счетчика производительности
+        /// </summary>
+        public override long CurrentValue
+        {
+            get
+            {
+                var counterCpy = _winCounter;
+                if (counterCpy == null)
+                    return Counter.FailureNum;
+
+                return counterCpy.RawValue;

```

Comment:
This will return the raw counter value (not the delta from the previous measurement).
The simpliest solution would be to do the same calculations as in ```InternalDeltaCounter```: 
```
        public override long CurrentValue
        {
            get
            {
                var counterCpy = _winCounter;
                if (counterCpy == null)
                    return Counter.FailureNum;


                long currentValue = counterCpy.RawValue;
                long lastMeasuredValue = Interlocked.Read(ref _lastMeasuredValue);
                return currentValue - lastMeasuredValue;
            }
        }
```

```_lastMeasuredValue``` field should be added to the class and should be updated inside ```Measure``` method.

Reply:
The same reodering update as in ```InternalDeltaCounter``` required here:
```C#
        public override long CurrentValue
        {
            get
            {
                var counterCpy = _winCounter;
                if (counterCpy == null)
                    return Counter.FailureNum;

                long lastMeasuredValue = Interlocked.Read(ref _lastMeasuredValue);
                long currentValue = counterCpy.RawValue;
                return currentValue - lastMeasuredValue;
            }
        }
```

Reply:
ok.